### PR TITLE
Add doc mapping UID attribute to `DocMapping`

### PR DESCRIPTION
**Pull request by guilload**
### Description
Per title.

### How was this PR tested?
Fixed and added backward compatibility tests.

### CODE REVIEW

File: quickwit/quickwit-doc-mapper/src/doc_mapping.rs [hunk 1]
```diff
@@ -99,6 +100,10 @@ impl Default for Mode {
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize, utoipa::ToSchema)]
 #[serde(deny_unknown_fields)]
 pub struct DocMapping {
+    /// UID of the doc mapping.
+    #[serde(default = "DocMappingUid::new")]
```
Comment (guilload) [inline quickwit/quickwit-doc-mapper/src/doc_mapping.rs:104]: This is going to cause an issue when deserializing doc mapping without an UID (we want to them to be 0). I'm working on a fix.

File: quickwit/quickwit-control-plane/src/control_plane.rs [hunk 2]
```diff
@@ -1681,10 +1681,12 @@ mod tests {
             .unwrap();
         assert!(indexing_tasks.is_empty());
 
-        let results: Vec<ApplyIndexingPlanRequest> =
-            client_inbox.drain_for_test_typed::<ApplyIndexingPlanRequest>();
-        assert_eq!(results.len(), 1);
-        assert!(results[0].indexing_tasks.is_empty());
+        let apply_plan_requests = client_inbox.drain_for_test_typed::<ApplyIndexingPlanRequest>();
+        assert!(!apply_plan_requests.is_empty());
```
Comment (trinity-1686a) [inline quickwit/quickwit-control-plane/src/control_plane.rs:1685]: was the test too strict in some instances?

File: quickwit/quickwit-doc-mapper/src/doc_mapping.rs [hunk 3]
```diff
@@ -99,6 +100,13 @@ impl Default for Mode {
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize, utoipa::ToSchema)]
 #[serde(deny_unknown_fields)]
 pub struct DocMapping {
+    /// Doc mapping UID.
+    ///
+    /// Splits with the same doc mapping UID share the same schema and should use the same doc
+    /// mapper during indexing and querying.
```
Comment (trinity-1686a) [inline quickwit/quickwit-doc-mapper/src/doc_mapping.rs:106]: am i right understanding that splits actually do not contain a doc mapping uid yet?

**Review by trinity-1686a — APPROVED** [review]
lgtm, just small questions

File: quickwit/quickwit-doc-mapper/src/doc_mapping.rs (hunk 3 above)
Comment (guilload) [reply quickwit/quickwit-doc-mapper/src/doc_mapping.rs:106]: Yes

File: quickwit/quickwit-control-plane/src/control_plane.rs (hunk 2 above)
Comment (guilload) [reply quickwit/quickwit-control-plane/src/control_plane.rs:1685]: Yes, sometimes the actor has time to spin more than once, and two messages are emitted, which causes the test to be flaky.

File: quickwit/quickwit-config/src/index_config/serialize.rs [hunk 4]
```diff
@@ -75,7 +80,7 @@ pub fn load_index_config_update(
     let current_index_parent_dir = &current_index_config
         .index_uri
         .parent()
-        .context("Unexpected `index_uri` format on current configuration")?;
+        .expect("index URI should have a parent");
```
Comment (rdettai) [inline quickwit/quickwit-config/src/index_config/serialize.rs:83]: we should not panic here

**Pull request merged**